src/test/java/uk/gov/ons/census/casesvc/service/NewAddressReportedServiceTest.java

```diff
@@ -0,0 +1,233 @@
+    collectionCase.setAddress(address);
+
+    NewAddress newAddress = new NewAddress();
+    newAddress.setCollectionCase(collectionCase);
+
+    PayloadDTO payloadDTO = new PayloadDTO();
+    payloadDTO.setNewAddress(newAddress);
+    ResponseManagementEvent newAddressEvent = new ResponseManagementEvent();
+    newAddressEvent.setPayload(payloadDTO);
+    EventDTO eventDTO = new EventDTO();
+    eventDTO.setDateTime(OffsetDateTime.now().minusSeconds(10));
+    newAddressEvent.setEvent(eventDTO);
+
+    Case casetoEmit = new Case();
+    when(caseService.saveNewCaseAndStampCaseRef(any(Case.class))).thenReturn(casetoEmit);
+    OffsetDateTime expectedDateTime = OffsetDateTime.now();
+
+    // When
+    underTest.processNewAddress(newAddressEvent, expectedDateTime);
+
+    // Then
+    Case expectedCase = getExpectedCase(collectionCase);
+
+    ArgumentCaptor<Case> caseArgumentCaptor = ArgumentCaptor.forClass(Case.class);
+    verify(caseService).saveNewCaseAndStampCaseRef(caseArgumentCaptor.capture());
+    Case actualCase = caseArgumentCaptor.getValue();
+    assertThat(actualCase).isEqualToComparingFieldByFieldRecursively(expectedCase);
+
+    verify(caseService).emitCaseCreatedEvent(casetoEmit);
+
+    verify(eventLogger)
+        .logCaseEvent(
+            eq(casetoEmit),
+            eq(eventDTO.getDateTime()),
+            eq("New Address reported"),
+            eq(EventType.NEW_ADDRESS_REPORTED),
+            eq(eventDTO),
+            eq(JsonHelper.convertObjectToJson(newAddressEvent.getPayload())),
+            eq(expectedDateTime));
+  }
+
+  @Test
+  public void testMinimalGoodNewAddressDoesNotFailValidation() {
+    ResponseManagementEvent responseManagementEvent = getMinimalValidNewAddress();
+    underTest.processNewAddress(responseManagementEvent, OffsetDateTime.now());
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testMissingUUID() {
+    ResponseManagementEvent newAddressEvent = getMinimalValidNewAddress();
+    newAddressEvent.getPayload().getNewAddress().getCollectionCase().setId("Xdd344234");
+
+    try {
+      underTest.processNewAddress(newAddressEvent, OffsetDateTime.now());
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage())
+          .isEqualTo("Expected NewAddress CollectionCase Id to be a valid UUID, got: Xdd344234");
+      throw e;
+    }
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testBadCasetype() {
+    ResponseManagementEvent newAddressEvent = getMinimalValidNewAddress();
+    newAddressEvent.getPayload().getNewAddress().getCollectionCase().setCaseType("BadCaseType");
+
+    try {
+      underTest.processNewAddress(newAddressEvent, OffsetDateTime.now());
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage())
+          .isEqualTo("Unexpected newAddress CollectionCase caseType: BadCaseType");
+      throw e;
+    }
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testBadAddressLevel() {
```

Comment:
Not "bad". **Missing**

The requirements say "check the minimum mandatory fields are present"